source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -669,14 +743,26 @@ static int s_process_read_message(
         (unsigned long long)downstream_window - processed);
 
     return AWS_OP_SUCCESS;
+
+shutdown_channel:
+    if (secure_transport_handler->read_delayed_shutdown_task) {
+        if (shutdown_error_code != AWS_ERROR_SUCCESS) {
+            secure_transport_handler->read_delayed_shutdown_task->error = shutdown_error_code;
+        }
+        /* Schedule the task to continue the shutdown process. */
+        aws_channel_schedule_task_now(slot->channel, &secure_transport_handler->read_delayed_shutdown_task->task);
```

Comment:
I don't understand why we're scheduling a task to call aws_channel_slot_on_handler_shutdown_complete()? Why not just call it synchronously? We've already sent any remaining messages at this point